File: scotts_old_java_examples/TestStuff/BasicActivity36NavGraph/app/src/main/java/com/example/blankactivity36navgraph/FirstFragment.java
package com.example.blankactivity36navgraph;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;
import androidx.navigation.fragment.NavHostFragment;

public class FirstFragment extends Fragment {
    TextView txt;

    @Override
    public View onCreateView(
            LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState
    ) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_first, container, false);
    }

    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        txt = getView().findViewById(R.id.textview_first);

        view.findViewById(R.id.button_first).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                //NOT NEEDED WITH THE SAFEARGS
                //NavHostFragment.findNavController(FirstFragment.this)
                        //.navigate(R.id.action_FirstFragment_to_SecondFragment)
                FirstFragmentDirections.ActionFirstFragmentToSecondFragment action = FirstFragmentDirections.actionFirstFragmentToSecondFragment();
                action.setMessage(txt.getText().toString());
                Navigation.findNavController(view).navigate(action);
            }
        });
    }
}

File: chapter8/navigationdemo/app/src/main/java/com/example/navigationdemo/FirstFragment.kt
package com.example.navigationdemo

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.Navigation
import com.example.navigationdemo.databinding.FragmentFirstBinding


class FirstFragment : Fragment() {
    private var _binding: FragmentFirstBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        _binding = FragmentFirstBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.firstFragBtn1.setOnClickListener {
            val action: FirstFragmentDirections.MainToSecondFragment = FirstFragmentDirections.mainToSecondFragment()
            action.message1 = binding.firstFragInput.text.toString()
            Navigation.findNavController(it).navigate(action)
        }

        binding.firstFragBtn2.setOnClickListener {
            val action: FirstFragmentDirections.MainToThirdFragment = FirstFragmentDirections.mainToThirdFragment()
            action.message2 = binding.firstFragInput.text.toString()
            Navigation.findNavController(it).navigate(action)
        }

    }

    override fun onDestroy() {
        super.onDestroy()
        _binding = null
    }

}